src/base.rs

```diff
@@ -0,0 +1,290 @@
+//! Primitives for object and array transmutation.
+//!
+//! The functions in this module are very unsafe and their use is not
+//! recommended unless you _really_ know what you are doing.
+use crate::error::Error;
+use crate::guard::{Guard, SingleManyGuard, SingleValueGuard, PermissiveGuard};
+#[cfg(feature = "std")]
+use core::mem::forget;
+use core::mem::size_of;
+use core::slice;
+
+/// Convert a byte slice into a single instance of a `Copy`able type.
+///
+/// The byte slice must have at least enough bytes to fill a single instance of
+/// a type, extraneous data is ignored.
+///
+/// # Safety
+///
+/// - This function does not perform memory alignment checks. The beginning of
+/// the slice data must be properly aligned for accessing the value of type `T`.
+/// - The byte data needs to correspond to a valid `T` value.
+///
+/// Failure to fulfill any of the requirements above is dangerous and may eat
+/// your laundry.
+///
+/// # Errors
+///
+/// An error is raised if the slice does not have enough bytes for a single
+/// value `T`.
+///
+/// # Examples
+///
+/// ```
+/// # use safe_transmute::base::from_bytes;
+/// # include!("../tests/test_util/le_to_native.rs");
+/// # fn main() {
+/// // Little-endian
+/// unsafe {
+/// # /*
+///     assert_eq!(from_bytes::<u32>(&[0x00, 0x00, 0x00, 0x01])?, 0x0100_0000);
+/// # */
+///     assert_eq!(from_bytes::<u32>(&[0x00, 0x00, 0x00, 0x01].le_to_native::<u32>()).unwrap(), 0x0100_0000);
+/// }
+/// # }
+/// ```
+pub unsafe fn from_bytes<T>(bytes: &[u8]) -> Result<T, Error>
```

Comment:
Requirements in brackets, as opposed to a `where` clause

Occurs on:
  *	L46
  *	L91
  *	L139
  *	L233